polygon/sync/sync.go

```diff
@@ -150,39 +152,67 @@ func (s *Sync) onNewHeaderEvent(
 	return nil
 }
 
-func (s *Sync) onNewHeaderHashesEvent(
+func (s *Sync) onNewBlockHashesEvent(
 	ctx context.Context,
-	event EventNewHeaderHashes,
+	event EventNewBlockHashes,
 	ccBuilder CanonicalChainBuilder,
 ) error {
-	for _, headerHashNum := range event.NewHeaderHashes {
+	for _, headerHashNum := range event.NewBlockHashes {
 		if (headerHashNum.Number <= ccBuilder.Root().Number.Uint64()) || ccBuilder.ContainsHash(headerHashNum.Hash) {
 			continue
 		}
 
-		newHeaders, err := s.p2pService.FetchHeaders(
-			ctx,
-			headerHashNum.Number,
-			headerHashNum.Number+1,
-			event.PeerId,
-		)
+		newBlocks, _, err := s.fetchNewBlocks(ctx, headerHashNum.Number, headerHashNum.Number+1, event.PeerId)
 		if err != nil {
 			return err
 		}
 
-		newHeaderEvent := EventNewHeader{
-			NewHeader: newHeaders[0],
-			PeerId:    event.PeerId,
+		newBlockEvent := EventNewBlock{
+			NewBlock: newBlocks[0],
+			PeerId:   event.PeerId,
 		}
 
-		err = s.onNewHeaderEvent(ctx, newHeaderEvent, ccBuilder)
+		err = s.onNewBlockEvent(ctx, newBlockEvent, ccBuilder)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func (s *Sync) fetchNewBlocks(
```

Comment:
could we move this into the p2p service?

Reply:
✅